app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+
+    for item in people:
+        person = Person(item["name"], item["age"])
+        if "wife" in item:
+            wife_name = item["wife"]
+            if wife_name in Person.people:
+                person.wife = Person.people[wife_name]
+                person.wife.husband = person
+        if "husband" in item:
+            husband_name = item["husband"]
+            if husband_name in Person.people:
+                person.husband = Person.people[husband_name]
+                person.husband.wife = person
```

Comment:
why do u need nested condintions here? u can use `and` for this, also u can use `get ` metod to get wife/husband from dict)